app/models/group.rb

```diff
@@ -0,0 +1,96 @@
+# == Schema Information
+#
+# Table name: groups
+#
+#  id                       :integer          not null, primary key
+#  name                     :string(255)      not null
+#  slug                     :string(255)      not null
+#  bio                      :string(255)      default(""), not null
+#  about                    :text             default(""), not null
+#  avatar_file_name         :string(255)
+#  avatar_content_type      :string(255)
+#  avatar_file_size         :integer
+#  avatar_updated_at        :datetime
+#  cover_image_file_name    :string(255)
+#  cover_image_content_type :string(255)
+#  cover_image_file_size    :integer
+#  cover_image_updated_at   :datetime
+#  confirmed_members_count  :integer          default(0)
+#  created_at               :datetime
+#  updated_at               :datetime
+#  avatar_processing        :boolean
+#
+
+class Group < ActiveRecord::Base
+  extend FriendlyId
+  friendly_id :name, :use => [:slugged, :history]
+
+  # We don't need to callback if we're killing the group off, just delete all
+  # memberships.
+  has_many :members, foreign_key: :group_id,
+    class_name: 'GroupMember', dependent: :delete_all
+  has_many :stories
+
+  has_attached_file :avatar,
```

Comment:
The partitioning I believe is to avoid hundreds of thousands of files per directory (some filesystems choke on that) by limiting it to 999 per. Is this an issue on S3? Agree that that's a better scheme to remove original name though